WordPress/src/main/java/org/wordpress/android/ui/sitecreation/domains/SiteCreationDomainsAdapter.kt

```diff
@@ -47,6 +53,15 @@ class SiteCreationDomainsAdapter(private val uiHelpers: UiHelpers) : Adapter<Sit
         diffResult.dispatchUpdatesTo(this)
     }
 
+    @Suppress("ForbiddenComment")
+    override fun onViewRecycled(holder: SiteCreationDomainViewHolder<*>) {
+        if (holder is DomainComposeItemViewHolder) {
+            // TODO: Remove this for Compose 1.2.0-beta02+ and RecyclerView 1.3.0-alpha02+
+            holder.composeView.disposeComposition()
+        }
+        super.onViewRecycled(holder)
+    }
+
```

Comment:
@thomashorta I added you as reviewer to this PR but only to have someone experienced with `Compose` validate this commit: https://github.com/wordpress-mobile/WordPress-Android/pull/17998/commits/9a35659fec875d0b09ecab4795878e787391d4f2

@ParaskP7 I added you as reviewer to inform someone from `infrastructure` of what the above commit entails (please read the code comments).

P.S. You don't have to really look at anything else from this PR, feel free to skip to the above commit and the only thing relevant from there is the comments; 

**TL;DR:** I added Compose configuration which should be removed when we upgrade to `Compose 1.2.0-beta02+` and `RecyclerView 1.3.0-alpha02+`, because they provide better out-of-the-box support for Using Compose in a RecyclerView[^source].

[^source]: [Using Compose in a RecyclerView](https://medium.com/androiddevelopers/jetpack-compose-interop-using-compose-in-a-recyclerview-569c7ec7a583
)

Reply:
Oh great, thank you so much for doing so and for connecting the dots @ovitrif , super important! 🥇 

Btw, how about you referencing that on this #17563 issue too, I think it will help having that there, so that when I get unblock (by #17790), I won't forget about this detail and possible update `RecyclerView` to its latest stable version (or even `RC`, should it be needed).

1. Question (❓): Will updating to `Compose 1.2.0-beta02+` alone not be enough?
2. Question (❓): It seems that `RecyclerView` is already on [1.3.0-rc01](https://developer.android.com/jetpack/androidx/releases/recyclerview#recyclerview-1.3.0-rc01), but this release has been there for months now (`September 21, 2022`). I hope the production ready release for `1.3.0` gets out sooner than later? As such, and again, just to be explicit about it, will updating to `RecyclerView 1.3.0` alone not be enough?
3. Question (❓): Maybe we should include this `RecyclerView 1.3.0-alpha02+` update to become part of #17563 too, wdyt?

Reply:
@ParaskP7 Thanks a lot for following up on this with more feedback 🙇 !

I added [this comment](https://github.com/wordpress-mobile/WordPress-Android/issues/17563#issuecomment-1439830915) to issue #17563 👍 

To answer `1` and `2`, I'm not really sure if both dependencies versions should be higher but if we have to check when upgrading only one of them I'm willing to do [the "trick" explained](https://gist.github.com/arriolac/cf4966d6f02f5890f8ea0058831fcadf#file-itemrow-kt) in the [linked medium article](https://medium.com/androiddevelopers/jetpack-compose-interop-using-compose-in-a-recyclerview-569c7ec7a583) to count recompositions, which should answer this 👍 .

My answer to question `3` is to include this `RecyclerView` version 👍 . Should I add something for this to the issue?

Reply:
Great, thanks you for adding the comment there @ovitrif ! 🌟 

PS: I updated the description accordingly so that we don't forget this extra change when we update `Compose` at some point in the future.

> To answer 1 and 2, I'm not really sure if both dependencies versions should be higher but if we have to check when upgrading only one of them I'm willing to do [the "trick" explained](https://gist.github.com/arriolac/cf4966d6f02f5890f8ea0058831fcadf#file-itemrow-kt) in the [linked medium article](https://medium.com/androiddevelopers/jetpack-compose-interop-using-compose-in-a-recyclerview-569c7ec7a583) to count recompositions, which should answer this 👍 .

Awesome, this helps a lot and I'll revisit/test that when the time comes! 💯 

> My answer to question 3 is to include this RecyclerView version 👍 . Should I add something for this to the issue?

Coolio, noted, I think I agree, but no need to add something there, at least not yet. I'll revisit the `RecyclerView` update and the removal of `holder.composeView.disposeComposition()` upon completing the `Compose` updates. 🙏